README.md

```diff
@@ -220,13 +221,77 @@ Cloudant({account:me, password:password}, function(err, cloudant) {
 });
 ~~~
 
-After initialization, in general, callback functions receive three arguments:
+### Callback Signature
+
+Callback functions receive three arguments:
+
+* `err` - The _error_ (if any). For example, fetching a document that doesn't exist:
+
+```
+{ Error: deleted
+    at Object.clientCallback (/usr/src/app/node_modules/nano/lib/nano.js:248:15)
+    at Request._callback (/usr/src/app/node_modules/@cloudant/cloudant/lib/clientutils.js:154:11)
+    ...
+  name: 'Error',
+  error: 'not_found',
+  reason: 'deleted',
+  scope: 'couch',
+  statusCode: 404,
+  request:
+   { method: 'GET',
+     headers:
+      { 'content-type': 'application/json',
+        accept: 'application/json' },
+     uri: 'http://localhost:5984/_users/895c3440-42e7-11e8-b9b2-358fa5dee4a0' },
+  headers:
+  { 'x-couchdb-body-time': '0',
+    'x-couch-request-id': '1c16b2b81f',
+    'transfer-encoding': 'chunked',
+    etag: '"7Q4MT2X8W1RO3JQOLSA4KGMV7"',
+    date: 'Fri, 27 Apr 2018 08:49:26 GMT',
+    'content-type': 'application/json',
+    'cache-control': 'must-revalidate',
+    statusCode: 404,
+    uri: 'http://localhost:5984/_users/895c3440-42e7-11e8-b9b2-358fa5dee4a0' },
+  errid: 'non_200',
+  description: 'couch returned 404' }
+```
+
+The corresponding database `request`, `headers` and `statusCode` are also returned in the error.
+
+* `body` - The HTTP _response body_ (if no error). For example:
+
+```js
+cloudant.db.list(function(err, body, header) {
+    console.log(body);
+});
+```
+
+```
+[ '_replicator', '_users' ]
+```
+
+* `header` - The HTTP _response header_ (if no error). For example:
```

Comment:
It seems weird that we call this `header` and not `headers` since it is all the response headers. Even if they are in a single object. WDYT?

Reply:
Yeah, agreed. I'll update all the instances in the README.